### PR TITLE
[KOGITO-5673] Setup BDD test for runtime with only events

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5673

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've ran `make before-pr` and everything is working accordingly
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by radtriste** [comment]
This is WIP

**Review by ricardozanini — APPROVED** [review]
Nice!

File: test/pkg/framework/kafka.go [hunk 1]
```diff
@@ -72,3 +76,89 @@ func ScaleKafkaInstanceDown(namespace, kafkaInstanceName string) error {
...
+
+// WaitForMessagesOnTopic waits for at least a certain number of messages are present on the given topic
+func WaitForMessagesOnTopic(namespace, kafkaInstanceName, topic string, numberOfMsg int, timeoutInMin int) error {
+	return WaitForOnOpenshift(namespace, fmt.Sprintf("%d message available on topic %s withing %d minutes", numberOfMsg, topic, timeoutInMin), timeoutInMin,
+		func() (bool, error) {
+			messages, err := GetMessagesOnTopic(namespace, kafkaInstanceName, topic)
+			if err != nil {
+				return false, err
+			}
+			GetLogger(namespace).Info(fmt.Sprintf("Got %d messages", len(messages)))
```
Comment (sutaakar) [inline test/pkg/framework/kafka.go:88]: Can you also print what messages were found? Can make a failure analysis easier.

File: test/pkg/framework/kafka.go [hunk 2]
```diff
@@ -72,3 +76,89 @@ func ScaleKafkaInstanceDown(namespace, kafkaInstanceName string) error {
...
+	}
+	GetLogger(namespace).Debug("Got kafka instance", "instance", kafkaInstance.Name)
+	bootstrapServer := infrastructure.ResolveKafkaServerURI(kafkaInstance)
+	if len(bootstrapServer) <= 0 {
+		GetLogger(namespace).Debug("Not able resolve URI for given kafka instance")
+		return nil, fmt.Errorf("not able resolve URI for given kafka instance %s", kafkaInstance.Name)
+	}
+	GetLogger(namespace).Debug("Got bootstrapServer", "server", bootstrapServer)
+
+	args := []string{"run", "-ti", "kafka-consumer", "--restart=Never"}
```
Comment (sutaakar) [inline test/pkg/framework/kafka.go:107]: Is it needed to start a new pod to connect to the bootstrap server? Can't we retrieve it from existing Kafka pod using `kubectl exec`?

Comment (radtriste) [reply test/pkg/framework/kafka.go:107]: right ... gonna update

Comment (radtriste) [reply test/pkg/framework/kafka.go]: updated !

File: test/pkg/framework/kafka.go (hunk 1 above)
Comment (radtriste) [reply test/pkg/framework/kafka.go:93]: added

**Review by ricardozanini — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]
Looks good to me.

**Comment by radtriste** [comment]
BDD failure is not related to this change (see https://issues.redhat.com/browse/KOGITO-5688)

**Pull request merged**